actions/sync-calendar/gcal/gcal.go

```diff
@@ -0,0 +1,118 @@
+// Package gcal handles Google Calendar integration.
+package gcal
+
+import (
+	"context"
+	"fmt"
+	"github.com/spinnaker/rotation-scheduler/proto/schedule"
+	"golang.org/x/oauth2/google"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	"google.golang.org/api/calendar/v3"
+	"google.golang.org/api/option"
+)
+
+const (
+	UserAgent  = "github.com/spinnaker/rotation-scheduler"
+	DateFormat = "2006-01-02"
+)
+
+// GCal wraps the Google Calendar service.
+type GCal struct {
+	CalendarID string
+	svc        *calendar.Service
+}
+
+// NewGCal wraps the calendar specified using the client. CalendarID should be a user's primary calendar (not a shared,
+// calendar), because the Clear() method only works on primary calendars.
+func NewGCal(calendarID string, client *http.Client) (*GCal, error) {
+	if calendarID == "" {
+		return nil, fmt.Errorf("calendar ID cannot be empty")
+	}
+
+	svc, err := calendar.NewService(context.Background(), option.WithHTTPClient(client), option.WithUserAgent(UserAgent))
+	if err != nil {
+		return nil, fmt.Errorf("unable to create Calendar service: %v", err)
+	}
+
+	return &GCal{
+		CalendarID: calendarID,
+		svc:        svc,
+	}, nil
+}
+
+// Client creates an authenticated HTTP Client from the specified service account credentials. The service account
+// must have a Domain-wide Delegation to control user calendars. See
+// https://developers.google.com/admin-sdk/directory/v1/guides/delegation
+func Client(calendarID, jsonCredentialsPath string) (*http.Client, error) {
```

Comment:
Also, the client is never used outside of this file, so the caller needs to construct it then pass it to `NewGCal`.  If we do need to manually create a client, could the `NewGCal` function create one instead of forcing the caller to?

Reply:
Oh, maybe this is for testing...

Reply:
Right, it eventually became this so I could use a different HTTP client.

That being said, there's not a fantastic solution here. I found the HTTP replay library from this long GH thread on testable client libraries: https://github.com/googleapis/google-cloud-go/issues/592

From that old internal project, we had mocks for each Cloud service we depended on. We definitely overused Mocks there, so I was seeking a different approach here. I'm not sold on the replay package either, because the actual interesting bits of the data (and those that may change in the future) are all base64 encoded.

Happy to discuss other options though.

Reply:
Ok, sounds good, you've obviously looked a lot more into it than me...it initially looked like we could replace all this with a simple `withCredentialsFile` but I see now the advantage of having an injectable client.